Release/src/pplx/pplx.cpp

```diff
@@ -63,58 +64,34 @@ static struct _pplx_g_sched_t
 {
     typedef std::shared_ptr<pplx::scheduler_interface> sched_ptr;
 
-    _pplx_g_sched_t() { m_state = post_ctor; }
-
-    ~_pplx_g_sched_t() { m_state = post_dtor; }
-
     sched_ptr get_scheduler()
-    {
-        switch (m_state)
+    {        
+        sched_ptr sptr = std::atomic_load_explicit(&m_scheduler, std::memory_order_consume);
+        if (!sptr)
         {
-            case post_ctor:
-                // This is the 99.9% case.
-
-                if (!m_scheduler)
-                {
-                    ::pplx::details::_Scoped_spin_lock lock(m_spinlock);
-                    if (!m_scheduler)
-                    {
-                        m_scheduler = std::make_shared<::pplx::default_scheduler_t>();
-                    }
-                }
-
-                return m_scheduler;
-            default:
-                // This case means the global m_scheduler is not available.
-                // We spin off an individual scheduler instead.
-                return std::make_shared<::pplx::default_scheduler_t>();
+            ::pplx::details::_Scoped_spin_lock lock(m_spinlock);
```

Comment:
As far as I am aware, all the major shared_ptr implementations actually use spinlocks to implement this load operation anyway; perhaps we should just drop the check and take the spinlock every time. It does limit scalability of the system but this shared_ptr city wasn't going to win awards for that anyway.

Reply:
Am I right that you suggest to drop double-lock here and just acquire spinlock on every call to `get_scheduler()`?

Reply:
@dimarusyy Yes, that's what my alternative does. Does that solution look acceptable to you?

Reply:
I'm fine as you accepted spinlock acquiring on every call to `get_scheduler()`.  Thanks for comment and update.